qa: Failure in feature_txindex_compatibility.py on master

https://api.cirrus-ci.com/v1/task/6143696841211904/logs/ci.log:
```
204/225 - [1mfeature_txindex_compatibility.py[0m failed, Duration: 1 s

[1mstdout:
[0m2021-11-15T10:09:01.666000Z TestFramework (INFO): Initializing test directory /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18
2021-11-15T10:09:02.259000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
    self.run_test()
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_txindex_compatibility.py", line 47, in run_test
    mini_wallet.send_self_transfer(from_node=self.nodes[1], utxo_to_spend=spend_utxo)
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/wallet.py", line 150, in send_self_transfer
    tx = self.create_self_transfer(**kwargs)
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/wallet.py", line 202, in create_self_transfer
    assert_equal(mempool_valid, tx_info['allowed'])
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 50, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(True == False)
2021-11-15T10:09:02.314000Z TestFramework (INFO): Stopping nodes
2021-11-15T10:09:02.571000Z TestFramework (WARNING): Not cleaning up dir /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18
2021-11-15T10:09:02.571000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18/test_framework.log
2021-11-15T10:09:02.572000Z TestFramework (ERROR): 
2021-11-15T10:09:02.572000Z TestFramework (ERROR): Hint: Call /tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/combine_logs.py '/tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20211115_095356/feature_txindex_compatibility_18' to consolidate all logs
2021-11-15T10:09:02.572000Z TestFramework (ERROR): 
2021-11-15T10:09:02.572000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2021-11-15T10:09:02.572000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2021-11-15T10:09:02.573000Z TestFramework (ERROR): 
```

--- a/test/functional/rpc_txoutproof.py
+++ b/test/functional/rpc_txoutproof.py
@@ -53,7 +53,7 @@ class MerkleBlockTest(BitcoinTestFramework):
         assert_equal(self.nodes[0].verifytxoutproof(self.nodes[0].gettxoutproof([txid1, txid2])), txlist)
         assert_equal(self.nodes[0].verifytxoutproof(self.nodes[0].gettxoutproof([txid1, txid2], blockhash)), txlist)
 
-        txin_spent = miniwallet.get_utxo()  # Get the change from txid2
+        txin_spent = miniwallet.get_utxo(txid=txid2)  # Get the change from txid2
         tx3 = miniwallet.send_self_transfer(from_node=self.nodes[0], utxo_to_spend=txin_spent)
         txid3 = tx3['txid']
         self.generate(self.nodes[0], 1)

--- a/test/functional/test_framework/wallet.py
+++ b/test/functional/test_framework/wallet.py
@@ -133,10 +133,9 @@ class MiniWallet:
 
         Args:
         txid: get the first utxo we find from a specific transaction
-
-        Note: Can be used to get the change output immediately after a send_self_transfer
         """
         index = -1  # by default the last utxo
+        self._utxos = sorted(self._utxos, key=lambda k: (k['value'], -k['height']))  # Put the largest utxo last
         if txid:
             utxo = next(filter(lambda utxo: txid == utxo['txid'], self._utxos))
             index = self._utxos.index(utxo)
@@ -172,8 +171,7 @@ class MiniWallet:
 
     def create_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None, mempool_valid=True, locktime=0, sequence=0):
         """Create and return a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
-        self._utxos = sorted(self._utxos, key=lambda k: (k['value'], -k['height']))
-        utxo_to_spend = utxo_to_spend or self._utxos.pop()  # Pick the largest utxo (if none provided) and hope it covers the fee
+        utxo_to_spend = utxo_to_spend or self.get_utxo()
         if self._priv_key is None:
             vsize = Decimal(104)  # anyone-can-spend
         else:

